valohai_yaml/objs/parameter_widget.py

```diff
@@ -0,0 +1,28 @@
+from collections import OrderedDict
+from typing import Optional, Union
+
+from valohai_yaml.objs.base import Item
+
+
+class ParameterWidget(Item):
+    """An UI widget for editing parameter values."""
+
+    def __init__(
+        self,
+        *,
+        type: str,
+    ) -> None:
+        self.type = type.lower()
+
+    @classmethod
+    def cast(cls, value: Optional[Union['ParameterWidget', OrderedDict, dict, str]]) -> Optional['ParameterWidget']:
```

Comment:
Also, ah, `cast` is only a thing for our `Enum`s; this should be `parse()` (and you should also define `serialize()`, so ParameterWidgets with just a type can be serialized back to a simple `str`.